driver-core/src/main/java/com/datastax/driver/core/ReplicationFactor.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.driver.core;
+
+// This class is a subset of server version at org.apache.cassandra.locator.ReplicationFactor
+
+import com.google.common.base.Preconditions;
+
+class ReplicationFactor {
+  private final int allReplicas;
+  private final int fullReplicas;
+  private final int transientReplicas;
+
+  ReplicationFactor(int allReplicas, int transientReplicas) {
+    this.allReplicas = allReplicas;
+    this.transientReplicas = transientReplicas;
+    this.fullReplicas = allReplicas - transientReplicas;
+  }
+
+  ReplicationFactor(int allReplicas) {
+    this(allReplicas, 0);
+  }
+
+  int fullReplicas() {
+    return fullReplicas;
+  }
+
+  int transientReplicas() {
+    return transientReplicas;
+  }
+
+  boolean hasTransientReplicas() {
+    return transientReplicas > 0;
+  }
+
+  static ReplicationFactor fromString(String s) {
+    if (s.contains("/")) {
+
+      int slash = s.indexOf('/');
+      String allPart = s.substring(0, slash);
+      String transientPart = s.substring(slash + 1);
+      Preconditions.checkArgument(
+          allPart != null && transientPart != null,
+          "Replication factor format is <replicas> or <replicas>/<transient>");
+      return new ReplicationFactor(Integer.parseInt(allPart), Integer.parseInt(transientPart));
+    } else {
+      return new ReplicationFactor(Integer.parseInt(s), 0);
+    }
+  }
+
+  @Override
+  public String toString() {
+    return allReplicas + (hasTransientReplicas() ? "/" + transientReplicas() : "");
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof ReplicationFactor)) {
+      return false;
+    }
+    ReplicationFactor that = (ReplicationFactor) o;
+    return allReplicas == that.allReplicas && transientReplicas == that.transientReplicas;
+  }
+
+  @Override
+  public int hashCode() {
+    return allReplicas ^ transientReplicas;
```

Comment:
That's a fancy hash code, shouldn't we go for the more mainstream ` Objects.hash(allReplicas, fullReplicas)`?

Reply:
I did use that at first, but there is a JDK6 check in the compile for 3.x and `Objects.hash()` was introduced in JDK7, so it errors the compile with this:
```
[INFO] --- animal-sniffer-maven-plugin:1.15:check (check-jdk6) @ cassandra-driver-core ---
[INFO] Checking unresolved references to org.codehaus.mojo.signature:java16:1.0
[ERROR] /home/emerkle/workspace/java-driver-3.x/driver-core/src/main/java/com/datastax/driver/core/ReplicationFactor.java:85: Undefined reference: int java.util.Objects.hash(Object[])
```

So I went with the old-school hash computation of XOR'ing the 2 attributes used to construct the object.

Reply:
Ah right, thanks for the explanation.